scripts/assess_detections.py

```diff
@@ -387,6 +368,48 @@ def main(cfg_file_path):
             .to_file(file_to_write, driver='GPKG', index=False)
         written_files.append(file_to_write)
 
+        # Save the metrics by class
+        metrics_by_cl_df = pd.DataFrame()
+        for dst in metrics_cl_df_dict.keys():
+            dst_df = metrics_cl_df_dict[dst].copy()
+            dst_thrsld_df = dst_df[dst_df.threshold==selected_threshold].copy()
+            dst_thrsld_df['dataset'] = dst
+            dst_thrsld_df.drop(columns=['threshold'], inplace=True)
+
+            metrics_by_cl_df = pd.concat([metrics_by_cl_df, dst_thrsld_df], ignore_index=True)
+        
+        metrics_by_cl_df['category'] = [
+            categories_info_df.loc[categories_info_df.label_class==det_class+1, 'CATEGORY'].iloc[0] 
+            for det_class in metrics_by_cl_df['class'].to_numpy()
+        ] 
+
+        file_to_write = os.path.join(OUTPUT_DIR, 'metrics_by_class.csv')
+        metrics_by_cl_df[
+            ['class', 'category', 'TP_k', 'FP_k', 'FN_k', 'precision_k', 'recall_k', 'dataset']
+        ].sort_values(by=['class', 'dataset']).to_csv(file_to_write, index=False)
+        written_files.append(file_to_write)
+
+        # Save the confusion matrix
```

Comment:
I am not familiar with the confusion matrix but I feel there is an issue with the label of the output.
Is the background not supposed to be named "missed labels" (or vice-versa)?
![image](https://github.com/swiss-territorial-data-lab/object-detector/assets/77794618/9a364ffc-e54a-49f1-ac48-05d173f27e23)
What is the plus value of adding this feature compared to detection tag and metrics calculation?

Reply:
In the sklearn version, the name at the tops are the predicted class and the ones in column are the actual class, like on the [Wikipedia page](https://en.wikipedia.org/wiki/Confusion_matrix#Example). I also find it quite counter intuitive, but I don't know how to signal it better in the table.
Would you think necessary to transpose the table?

Reply:
When you have only one class, there are no real advantage since it can easily be deduced from the P and R. When you have several classes. It really helps understand what got lost where.
I can modify the code if you prefer so the confusion matrix is exported only in the multi-class case.

Reply:
In the case with one class, like the one presented here, it would be for the false positives. So the ones detected as "quarries" but that are actually "background".